cmd/metal-api/internal/scaler/poolscaler_test.go

```diff
@@ -0,0 +1,160 @@
+package scaler
+
+import (
+	"testing"
+
+	"github.com/metal-stack/metal-api/cmd/metal-api/internal/metal"
+	"go.uber.org/zap/zaptest"
+)
+
+func TestPoolScaler_AdjustNumberOfWaitingMachines(t *testing.T) {
+	tests := []struct {
+		name      string
+		partition *metal.Partition
+		mockFn    func(mock *MockMachineManager)
+		wantErr   bool
+	}{
+		{
+			name: "waiting machines match pool size; do nothing",
+			partition: &metal.Partition{
+				WaitingPoolMinSize: "10",
+				WaitingPoolMaxSize: "20",
+			},
+			mockFn: func(mock *MockMachineManager) {
+				mock.On("WaitingMachines").Once().Return(append(metal.Machines{}, make([]metal.Machine, 10)...), nil)
+				mock.On("AllMachines").Once().Return(append(metal.Machines{}, make([]metal.Machine, 150)...), nil)
+			},
+			wantErr: false,
+		},
+		{
+			name: "waiting machines match pool size in percent; do nothing",
+			partition: &metal.Partition{
+				WaitingPoolMinSize: "25%",
+				WaitingPoolMaxSize: "30%",
+			},
+			mockFn: func(mock *MockMachineManager) {
+				mock.On("AllMachines").Once().Return(append(metal.Machines{}, make([]metal.Machine, 150)...), nil)
+				mock.On("WaitingMachines").Once().Return(append(metal.Machines{}, make([]metal.Machine, 45)...), nil)
+			},
+			wantErr: false,
+		},
+		{
+			name: "more waiting machines needed; power on 8 machines",
+			partition: &metal.Partition{
+				WaitingPoolMinSize: "15",
+				WaitingPoolMaxSize: "20",
+			},
+			mockFn: func(mock *MockMachineManager) {
+				mock.On("AllMachines").Once().Return(append(metal.Machines{}, make([]metal.Machine, 150)...), nil)
+				mock.On("WaitingMachines").Once().Return(append(metal.Machines{}, make([]metal.Machine, 10)...), nil)
+				mock.On("ShutdownMachines").Once().Return(append(metal.Machines{}, make([]metal.Machine, 10)...), nil)
+				mock.On("PowerOn", &metal.Machine{}).Return(nil).Times(8)
+			},
+			wantErr: false,
+		},
+		{
+			name: "more waiting machines needed in percent; power on 10 machines",
+			partition: &metal.Partition{
+				WaitingPoolMinSize: "30%",
+				WaitingPoolMaxSize: "40%",
+			},
+			mockFn: func(mock *MockMachineManager) {
+				mock.On("AllMachines").Once().Return(append(metal.Machines{}, make([]metal.Machine, 150)...), nil)
+				mock.On("WaitingMachines").Once().Return(append(metal.Machines{}, make([]metal.Machine, 35)...), nil)
+				mock.On("ShutdownMachines").Once().Return(append(metal.Machines{}, make([]metal.Machine, 20)...), nil)
+				mock.On("PowerOn", &metal.Machine{}).Return(nil).Times(18)
```

Comment:
Isn't 10 machines expected? There is also the possibility to call `manager.AssertExpectations(t)` at the end of a test to make sure the correct amount of calls was made (e.g. when a call was not made).

Reply:
I forgot to update the name of the test. 18 machines are expected:  
```
min = 0.3*150 = 45
max = 0.4*150 = 60
average = round(52.5) = 53
waiting = 35
waiting - average = -18
```

Is calling `manager.AssertExpectations(t)` really necessary? My tests were always failing, if an incorrect number of calls were made.